docs/design/mono/diagnostics-tracing.md

```diff
@@ -0,0 +1,92 @@
+# MonoVM Diagnostics Tracing Component
+
+## Summary
+
+MonoVM includes support for EventPipe and DiagnosticServer components used to generate nettrace files including both runtime as well as custom `EventSource` events. It is possible to either use dynamic components (Android) or link statically (iOS) depending on build configuration. EventPipe will mainly be used during development/testing cycle, and should not be deployed or linked into builds passed to app store for verification and publication.
+
+## Scenarios
+
+.Net supports a number of different EventPipe scenarios using tools mainly from diagnostics repository. MonoVM include support for several of these scenarios, running tools like `dotnet-counters` and `dotnet-trace` to collect and analyze runtime performance data. Other things like requesting a core/gc dump or attaching profiler over EventPipe is currently not supported on MonoVM.
```

Comment:
will **core/gc dump or attaching profiler** be supported when .net6 release?

Reply:
No, since format is different between runtimes and existing tooling to analyse the data (SOS) is currently not supporting Mono runtime. Primary focus for .net6 is to get EventPipe up and running on Mono for mobile (Android/iOS) supporting things like dotnet-trace, dotnet.counters scenarios, custom EventSource, custom counters and inproc EventListeners.

Reply:
ok, but how can I profiler mono gc problem ?

Reply:
Until we have more features wired up into EventPipe, regular mono profiler works as is.